File: src/main/java/com/example/SpringBoot/LeetCood/ContainerWithMostWater/ContainerWithMostWater.java
package com.example.SpringBoot.LeetCood.ContainerWithMostWater;


/**
 * leetcood题目：盛水最多的容器
 *
 * @author duwenxu
 * @create 2019-04-02 16:27
 */
public class ContainerWithMostWater {
    public static int solution(int[] height) {
        int maxArea = 0;
        for (int i = 0; i < height.length; i++) {
            for (int j = i + 1; j < height.length; j++) {
                maxArea = Math.max(maxArea, Math.min(height[i], height[j]) * (j - i));
            }
        }
        return maxArea;
    }


    public static void main(String[] args) {
       int[] arr={1,8,6,2,5,4,8,3,7};
        System.out.println(solution(arr));
    }
}

File: src/main/java/com/example/SpringBoot/LeetCood/ContainerWithMostWater/ContainerWithMostWater.kt
package com.example.SpringBoot.LeetCood.ContainerWithMostWater

/**
* @Description: 盛水最多的容器
* @Date: 2019/3/25
 *
 * 思想：指针对撞  只进行一次扫描
*/
class Solution{
    fun maxArea(height:IntArray):Int{
        var area=0
        var l=0
        var r=height.size-1
        while (l<r){
            area=Math.max(area,Math.min(height[l],height[r])*(r-l))
            when(height[l]<height[r]){
                true->l++
                else->r--
            }
        }
        return area
    }
}